agent/base

```diff
@@ -117,7 +117,7 @@ if [[ -z "${_pbench_full_hostname}" ]]; then
 fi
 validate_hostname "${_pbench_full_hostname}" "_pbench_full_hostname"
 if [[ -z "${_pbench_hostname_ip}" ]]; then
-    export _pbench_hostname_ip=$(hostname -i)
+    export _pbench_hostname_ip=$(hostname -i | awk '{print $1}')
```

Comment:
I had to do a `man` to see exactly what `hostname -i` is defined to do. Interestingly, it warns _Display  the  network  address(es)  of the host name. __Note that this works only if the host name can be resolved. Avoid using this option__; use hostname --all-ip-addresses instead._

This doesn't depend on DNS resolution, and I wonder if that'd be better in a cloud virtual deployment that might not have DNS?

Reply:
You can make this happen by modifying `/etc/hosts` to have an alternate IP for localhost or your official DHCP host name.  Then `hostname -i` returns N IP addresses.